### ISSUE TITLE
Problem Build Msys2/Mingw32 on Windows

### ISSUE THREAD

**Issue by MartyShepard** [diasurgical/devilutionX#781]
After a Clean Msys2 Install &  Configure  it show a Missing Separator

# cmake -DCMAKE_TOOLCHAIN_FILE=../CMake/mingwcc.cmake ..
-- Building for: NMake Makefiles
-- The C compiler identification is GNU 10.2.0
-- The CXX compiler identification is GNU 10.2.0
-- Check for working C compiler: D:/System_Development/MSYS2_devilutionX/mingw32/bin/i686-w64-mingw32-gcc.exe
-- Check for working C compiler: D:/System_Development/MSYS2_devilutionX/mingw32/bin/i686-w64-mingw32-gcc.exe - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: D:/System_Development/MSYS2_devilutionX/mingw32/bin/i686-w64-mingw32-g++.exe
-- Check for working CXX compiler: D:/System_Development/MSYS2_devilutionX/mingw32/bin/i686-w64-mingw32-g++.exe - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found sodium: D:/System_Development/MSYS2_devilutionX/mingw32/lib/libsodium.a (found version "1.0.18")
-- Found SDL2_ttf: D:/System_Development/MSYS2_devilutionX/mingw32/lib/libSDL2_ttf.dll.a
-- Found SDL2: D:/System_Development/MSYS2_devilutionX/mingw32/lib/libSDL2.dll.a
-- Found SDL2_mixer: D:/System_Development/MSYS2_devilutionX/mingw32/lib/libSDL2_mixer.dll.a
-- Configuring done
-- Generating done
-- Build files have been written to: D:/System_Development/MSYS2_devilutionX/src/devilutionX/build

# make
Makefile:38: *** Fehlender Trenner.  Schluss.

regards

**Comment by AJenbo** [comment]
> Makefile: 38: *** Missing separator. Closing.

Sorry, i don't have experience with Msys2 (and only limited understanding of German). Let us know if you find a solution or know what the issue could be related to.

**Comment by MartyShepard** [comment]
Hello. No Problem :)

I have not very experience with makefiles but if id do a TAB in the build\makefile on 

```
!IF "$(OS)" == "Windows_NT"
NULL=
!ELSE
NULL=nul
!ENDIF
SHELL = cmd.exe 
```

to
```
	!IF "$(OS)" == "Windows_NT"
	NULL=
	!ELSE
	NULL=nul
	!ENDIF
	SHELL = cmd.exe
```
it works


other Problems before was on the File mingwcc.cmake at generating the Configuration

My Workspace ist on 

`D:/System_Development/MSYS2_devilutionX/`

At build the Cmake Configuration it it redirect and try to find Sodium, SDL2 and SDL2 Stuff then 
on.

`D:\usr\i686-w64-mingw32\i686-w64-mingw32\System_Development\MSYS2_devilutionX\i686-w64-mingw32\i686-w64-mingw32\include\sodium.h"

thic can i avoid if i remove in the mingwcc.cmake

`list(APPEND CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES "${CROSS_PREFIX}/i686-w64-mingw32/include")`

and

`SET(CMAKE_FIND_ROOT_PATH  "${CROSS_PREFIX}/i686-w64-mingw32" "${CROSS_PREFIX}/i686-w64-mingw32/i686-w64-mingw32")`

Now it find all files in the current MSYS2-MingW Directory

Other thing is. The MingW Configuation build need nMake from Visualstudio. So i had Manual copy the nMake.exe from my vmware HD with Visual Studio to the Msys2-MingW32 bin directory.

more it brings a CMAKE_AR-NOTFOUND.exe but ar.exe is in bin. I copy and renamed bin\ar.exe it to bin\CMAKE_AR-NOTFOUND.exe

now it could Compile but breaks with 

make
Makefile:38: *** Missing Seperator. Stop. (the main Problem )

 
regards

**Comment by AJenbo** [comment]
Hmm this is all a bit Windows specific for me. Any chance you can build using WSL instead?

**Comment by MartyShepard** [comment]
Hmm... i do it on Windows7 Ultimate. My 2nd Computer with Windows10 is not yet ready.

...

**Comment by AJenbo** [comment]
Ok. Other options are MSVC 2017 (2019 has a compiler bug that affects the game) or a virtual machine. Maybe someone on the chat knows about Msys2.

**Comment by MartyShepard** [comment]
Hello,

I try this. :)

**Comment by ThomasChr** [comment]
@MartyShepard Is this still an Issue?

**Comment by ThomasChr** [comment]
No answer after nearly two month. @AJenbo should we close this?

**Comment by ludescher** [comment]
I encountered the same problem and eventually discovered some [information](https://www.cs.colby.edu/maxwell/courses/tutorials/maketutor/) that explains why it's not working.

> One very important thing to note is that there is a tab before the gcc command in the makefile. There must be a tab at the beginning of any command, and make will not be happy if it's not there.

**Comment by AJenbo** [comment]
That doesn't make sense to me, have you been able to make it work?

**Comment by ludescher** [comment]
I encountered a similar error (`Makefile:25: *** Fehlender Trenner. Schluss.`) in a different project, not the exact same issue discussed here. I apologize if I wasn’t clear about that initially. I stumbled upon this thread while searching for a solution and additional information, so I decided to share my findings in hopes that they might help someone else.

For me, the problem turned out to be due to using spaces instead of tabs in my Makefile. I don’t fully understand why the difference in whitespace mattered, but switching to tabs resolved the issue. As a note, I'm working on Linux/Debian, so the behavior might differ on Windows. I'm still quite new to Make, and I hope this explanation clarifies things rather than causing confusion.

I hope my insights are helpful, cheers!
